sqldelight-core/src/main/kotlin/com/squareup/sqldelight/core/compiler/model/BindableQuery.kt

```diff
@@ -87,6 +87,18 @@ open class BindableQuery(
       index to arg.copy(name = name)
     }
 
+    if (statement is InsertStmtMixin) {
+      return@lazy result.map { (index, argument) ->
+        val isPrimaryKey = argument.column?.columnConstraintList
+            ?.any { it.node?.findChildByType(SqliteTypes.PRIMARY) != null } == true
+        if (isPrimaryKey && argument.column?.typeName?.text == "INTEGER") {
```

Comment:
There's no constant to compare against?

Reply:
unfortunately no cause its a string but i can make a static constant in the core

Reply:
I would only do it if the string versions of SQL types are used in multiple places. Feel free to merge this as-is though.